server/graphql/document/src/types/document.rs

```diff
@@ -0,0 +1,112 @@
+use async_graphql::*;
+use async_graphql::{dataloader::DataLoader, Context};
+use chrono::{DateTime, Utc};
+
+use graphql_core::{standard_graphql_error::StandardGraphqlError, ContextExt};
+use repository::Document;
+use service::document::raw_document::RawDocument;
+
+use crate::json_schema_loader::JsonSchemaLoader;
+
+use super::json_schema::JSONSchemaNode;
+
+pub struct DocumentNode {
+    pub document: Document,
+}
+
+#[Object]
+impl DocumentNode {
+    pub async fn id(&self) -> &str {
+        &self.document.id
+    }
+
+    pub async fn name(&self) -> &str {
+        &self.document.name
+    }
+
+    pub async fn parents(&self) -> &[String] {
+        &self.document.parents
+    }
+
+    pub async fn author(&self) -> &str {
+        &self.document.author
+    }
+
+    pub async fn timestamp(&self) -> &DateTime<Utc> {
+        &self.document.timestamp
+    }
+
+    pub async fn r#type(&self) -> &str {
+        &self.document.r#type
+    }
+
+    pub async fn data(&self) -> &serde_json::Value {
+        &self.document.data
+    }
+
+    pub async fn schema(&self, ctx: &Context<'_>) -> Result<Option<JSONSchemaNode>> {
+        Ok(match &self.document.schema_id {
+            Some(schema_id) => {
+                let loader = DataLoader::new(
+                    JsonSchemaLoader {
+                        connection_manager: ctx.get_connection_manager().clone(),
+                    },
+                    async_std::task::spawn,
+                );
+                let schema = loader.load_one(schema_id.clone()).await?.ok_or(
+                    StandardGraphqlError::InternalError(format!(
+                        "Cannot find schema {}",
+                        schema_id
+                    ))
+                    .extend(),
+                )?;
+                Some(JSONSchemaNode { schema })
+            }
+            None => None,
+        })
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct DocumentConnector {
+    pub total_count: u32,
+    pub nodes: Vec<DocumentNode>,
+}
+
+pub struct RawDocumentNode {
+    pub document: RawDocument,
+}
+
+#[Object]
+impl RawDocumentNode {
```

Comment:
Why do we need to expose RawDocumentNode ?

Reply:
Its for the merge confligc, I didn't wanted to set the id yet because the FE could still decide to do some further edits which would invalidate the id.